bluesky/traffic/route.py

```diff
@@ -138,73 +155,71 @@ def addwptStack(self, idx, *args):  # args: all arguments of addwpt
                 if self.wpname[i].count("/") >0:
 #                    print (self.wpname[i])
                     rwyrteidx = i
-                i = i + 1
+                i += 1
 
             # Only TAKEOFF is specified wihtou a waypoint/runway
-            if len(args) == 1 or args[1] == "" or args[1] is None:
+            if len(args) == 1 or not args[1]:
                 # No runway given: use first in route or current position
 
                 # print ("rwyrteidx =",rwyrteidx)
                 # We find a runway in the route, so use it
                 if rwyrteidx>0:
-                    rwlat   = self.wplat[rwyrteidx]
-                    rwlon   = self.wplon[rwyrteidx]
-                    aptidx  = bs.navdb.getapinear(rwlat,rwlon)
+                    rwylat   = self.wplat[rwyrteidx]
+                    rwylon   = self.wplon[rwyrteidx]
+                    aptidx  = bs.navdb.getapinear(rwylat,rwylon)
                     aptname = bs.navdb.aptname[aptidx]
 
-                    if self.wpname[rwyrteidx].count("/")>1:
-                        rwname = self.wpname[rwyrteidx].split("/")[1]
-                    else:
-                        rwname = self.wpname[rwyrteidx]
-                    rwyid = rwname.replace("RWY","").replace("RW","")
+                    rwyname = self.wpname[rwyrteidx].split("/")[1]
+                    rwyid = rwyname.replace("RWY","").replace("RW","")
                     rwyhdg = bs.navdb.rwythresholds[aptname][rwyid][2]
 
                 else:
-                    rwlat  = bs.traf.lat[idx]
-                    rwlon  = bs.traf.lon[idx]
+                    rwylat  = bs.traf.lat[idx]
+                    rwylon  = bs.traf.lon[idx]
                     rwyhdg = bs.traf.trk[idx]
 
-            elif args[1].count("/") > 0 or not (args[2] == "" or args[2] is None): # we need apt,rwy
+            elif args[1].count("/") > 0 or len(args) > 2 and args[2]: # we need apt,rwy

```

Comment:
Need to put `len(args)>2` as otherwise an `IndexError` can happen.  Simplified `args[2]` condition.